react/formatAddressToString.ts

```diff
@@ -0,0 +1,43 @@
+import { Address } from 'vtex.checkout-graphql'
+import { CountryRules } from 'vtex.address-context/types'
+import msk from 'msk'
+
+export default function formatAddressToString(
+  address: Address,
+  rules: CountryRules
+) {
+  const displaySpec = rules.display.compact
+
+  return displaySpec
+    .map(line => line.filter(fragment => address[fragment.name]))
+    .filter(line => line.length > 0)
+    .map(line =>
+      line
+        .map((fragment, index) => {
+          const hasPreviousFragment = index > 0 && address[line[index - 1].name]
```

Comment:
I'm thinking about refactoring this function, `PlaceDetails` and `AddressForm` later because this "rendering" logic of the address is the same among all of them, they just differ on what is the output (for this function it's a string, for place details it's spans and for address form it's inputs)